cpu/xive.py

```diff
@@ -72,15 +82,34 @@ def test_storeEOI(self):
                           "missing %s / %s" % (self.hw, self.intr))
             if is_dir:
                 # If xive is a directory then read from store-eoi file
-                xive_path = "/sys/kernel/debug/powerpc/xive/store-eoi"
-
-            flags = genio.read_file(xive_path)
-            if is_dir:
-                # store-eoi value can be enabled or disable via kernel command
-                # line or using sysfs debug directory.
-                match = re.search("Y", flags) or re.search("N", flags)
+                store_eoi_file = os.path.join(xive_path, "store-eoi")
+                is_file_exists = os.path.exists(store_eoi_file)
+                if is_file_exists:
+                    flags = genio.read_file(store_eoi_file)
+
+                    # store-eoi value can be enabled or disable via kernel
+                    # command line or using sysfs debug directory.
+                    match = re.search("Y", flags) or re.search("N", flags)
+                else:
+                    if (self.detected_distro.name == 'rhel'):
+                        self.cancel("Test Skipped! Invalid scenario for RHEL")
+                    elif (self.detected_distro.name == 'SuSE'):
+                        interrupts_file = os.path.join(xive_path, "interrupts")
+                        ipis_file = os.path.join(xive_path, "ipis")
+
```

Comment:
We don't necessarily need to add checks for distro releases. We can check for which files are present and take appropriate action. Also it's not required to check for both ipis and interrupts file, only one will suffice.

We can define a new variable xive_file and assign the value based on which file should be used to read store-eoi flag information. We then use this variable and pass it to read_file() .. rest of the code then can remain unchanged

Reply:
Okay, noted. On it!

> We don't necessarily need to add checks for distro releases. We can check for which files are present and take appropriate action. Also it's not required to check for both ipis and interrupts file, only one will suffice.
> 
> We can define a new variable xive_file and assign the value based on which file should be used to read store-eoi flag information. We then use this variable and pass it to read_file() .. rest of the code then can remain unchanged

Reply:
> Okay, noted. On it!
> 
> > We don't necessarily need to add checks for distro releases. We can check for which files are present and take appropriate action. Also it's not required to check for both ipis and interrupts file, only one will suffice.
> > We can define a new variable xive_file and assign the value based on which file should be used to read store-eoi flag information. We then use this variable and pass it to read_file() .. rest of the code then can remain unchanged

@sacsant Done.